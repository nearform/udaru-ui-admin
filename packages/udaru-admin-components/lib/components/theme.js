// Adapted from default theme on https://github.com/jxnblk/styled-system

// breakpoint values
// any array length works with styled-system
const breakpoints = ['500px', '800px', '1200px']

const colors = {
  primary: '#337ab7',
  primaryActive: '#02A171',
  danger: '#d9534f',
  lightGrey: '#D8D8D8',
  lightGrey2: '#888',
  lightGrey3: '#eee',
  darkGrey: '#555',
  black: '#333',
  green: '#5cb85c',
  lightBlue: '#5bc0de',
  orange: '#f0ad4e'
}

export { colors }

export const radius = '4px'

export const maxContainerWidth = '1280px'

// boxShadows
export const boxShadows = [
  `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
]

export const speeds = {
  slow: '1s',
  medium: '.4s',
  fast: '.2s'
}

export const easings = {
  easeOut: 'cubic-bezier(0,.6,.3,1)',
  expoEaseOut: 'cubic-bezier(0,1,.3,1)',
  bounce: 'cubic-bezier(0,1.2,.5,1.4)'
}

// space is used for margin and padding scales
// it's recommended to use powers of two to ensure alignment
// when used in nested elements
// numbers are converted to px
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

// typographic scale
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 96, 128]

// for any scale, either array or objects will work
const lineHeights = [1, 1.125, 1.25, 1.5]

const fontWeights = {
  normal: 500,
  bold: 700
}

const letterSpacings = {
  normal: 'normal',
  caps: '0.25em'
}

// border-radius
const radii = [0, 2, 4, 8]

const borders = [0, '1px solid', '2px solid']

const shadows = [`0 1px 2px 0 ${colors.text}`, `0 1px 4px 0 ${colors.text}`]

const theme = {
  breakpoints,
  colors,
  space,
  fontSizes,
  lineHeights,
  fontWeights,
  letterSpacings,
  radii,
  radius,
  borders,
  shadows,
  boxShadows,
  speeds,
  easings
}

export default theme
