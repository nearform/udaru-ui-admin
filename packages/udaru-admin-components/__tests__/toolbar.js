import React from 'react'
import renderer from 'react-test-renderer'
import Toolbar from 'toolbar'
import { createSerializer } from 'jest-emotion'
import * as emotion from 'emotion'
import theme from '../lib/components/theme'

expect.addSnapshotSerializer(createSerializer(emotion))

it('should render with default props', () => {
  const component = renderer.create(<Toolbar theme={theme} />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('should render with disable props', () => {
  const component = renderer.create(
    <Toolbar
      theme={theme}
      disableView={false}
      disableUpdate={false}
      disableDelete={false}
    />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
