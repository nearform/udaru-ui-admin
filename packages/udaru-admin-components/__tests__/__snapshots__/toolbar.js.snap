// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={undefined}
          type="button"
        >
          New Team
        </button>
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "display": "inline-block",
              "padding": "0 10px",
            }
          }
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={undefined}
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            type="button"
          >
            View Team
          </button>
        </div>
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "display": "inline-block",
              "padding": "0 10px 0 0",
            }
          }
        >
          <button
            className="btn btn-info"
            disabled={true}
            onClick={undefined}
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            type="button"
          >
            Update Team
          </button>
        </div>
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <button
            className="btn btn-danger"
            disabled={true}
            onClick={undefined}
            style={
              Object {
                "pointerEvents": "none",
              }
            }
            type="button"
          >
            Delete Team
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with disable props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={undefined}
          type="button"
        >
          New Team
        </button>
        <div
          onClick={null}
          style={
            Object {
              "display": "inline-block",
              "padding": "0 10px",
            }
          }
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={undefined}
            style={Object {}}
            type="button"
          >
            View Team
          </button>
        </div>
        <div
          onClick={null}
          style={
            Object {
              "display": "inline-block",
              "padding": "0 10px 0 0",
            }
          }
        >
          <button
            className="btn btn-info"
            disabled={false}
            onClick={undefined}
            style={Object {}}
            type="button"
          >
            Update Team
          </button>
        </div>
        <div
          onClick={null}
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={undefined}
            style={Object {}}
            type="button"
          >
            Delete Team
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
